golem-worker-service-base/src/service/api_definition_validator.rs

```diff
@@ -11,8 +12,12 @@ use golem_service_base::model::{Export, ExportFunction, ExportInstance, Template
 use serde::{Deserialize, Serialize};
 
 #[async_trait]
-pub trait ApiDefinitionValidatorService {
-    async fn validate(&self, api: &ApiDefinition) -> Result<(), ValidationError>;
+pub trait ApiDefinitionValidatorService<AuthCtx> {
+    async fn validate(
+        &self,
+        api: &ApiDefinition,
+        auth_ctx: &AuthCtx,
```

Comment:
Instead of auth_ctx, you could take template-metadata. I believe, conceptually, that's a much reusable way of doing it.  If not now, later we would need it.

Reply:
i think it is about, that it require to call TemplateService, which require authorisation

but sure other solution, could be that as namespace is result of authorisation, namespace could also contain authorisation data, which can be used in that invocation

Reply:
This was implemented. Removed auth in favor of explicitly passing all templates to validator